src/modules/rover_pos_control/rover_pos_control_params.c

```diff
@@ -248,17 +249,26 @@ PARAM_DEFINE_FLOAT(GND_SPEED_TRIM, 3.0f);
 PARAM_DEFINE_FLOAT(GND_SPEED_MIN, 1.0f);
 
 /**
- * Maximum ground speed
+ * Maximum output of velocity PID controller
  *
+ * Used for limiting output of the speed PID controller (which is somewhat in m/s unit)
+ *
+ * Note: We need to figure out what the unit dimension is for the output of the Velocity PID controller,
+ * since we are mapping it directly to throttle, and it will be then *sqrt of thrust (for RPM controlled ESCs),
+ * but we are calculating based off of velocity error.
+ *
+ * Note: What is the role of the I-term? E.g. in Multicopter, the I-term of the Velocity controller accounts for
+ * constant offset force (acceleration) from external source. E.g. Wind.
+ *
+ * What is it for the boat?
  *
- * @unit m/s
  * @min 0.0
  * @max 40
  * @decimal 1
  * @increment 0.5
  * @group Rover Position Control
  */
-PARAM_DEFINE_FLOAT(GND_SPEED_MAX, 10.0f);
+PARAM_DEFINE_FLOAT(GND_SPEED_PID_MX, 10.0f);
```

Comment:
I renamed the param as it doesn't really describe well what was actually doing (this is NOT! the Maximum speed limit, it's a limit on velocity PID controller's output (which is more of an acceleration unit command))